compiler/AST/wellknown.cpp

```diff
@@ -200,28 +306,63 @@ std::vector<Type*> getWellKnownTypes()
 {
   std::vector<Type*> types;
 
-  int nEntries = sizeof(sWellKnownTypes) / sizeof(sWellKnownTypes[0]);
+  int nTypes = sizeof(sWellKnownTypes) / sizeof(sWellKnownTypes[0]);
+  int nAggregate = sizeof(sWellKnownAggregateTypes) /
+                   sizeof(sWellKnownAggregateTypes[0]);
 
-  for (int i = 0; i < nEntries; ++i) {
+  for (int i = 0; i < nTypes; i++) {
     WellKnownType& wkt = sWellKnownTypes[i];
     if (*wkt.type_ != NULL)
       types.push_back(*wkt.type_);
   }
+  for (int i = 0; i < nAggregate; i++) {
+    WellKnownAggregateType& wkt = sWellKnownAggregateTypes[i];
+    if (*wkt.type_ != NULL)
+      types.push_back(*wkt.type_);
+  }
 
   return types;
 }
 
 void clearGenericWellKnownTypes()
 {
-  int nEntries = sizeof(sWellKnownTypes) / sizeof(sWellKnownTypes[0]);
+  int nTypes = sizeof(sWellKnownTypes) / sizeof(sWellKnownTypes[0]);
+  int nAggregate = sizeof(sWellKnownAggregateTypes) /
+                   sizeof(sWellKnownAggregateTypes[0]);
 
-  for (int i = 0; i < nEntries; ++i) {
+  for (int i = 0; i < nTypes; i++) {
     WellKnownType& wkt = sWellKnownTypes[i];
+    Type* t = *wkt.type_;
+    if (t != NULL && t->symbol && t->symbol->hasFlag(FLAG_GENERIC))
```

Comment:
None of these should be generic, right?

Reply:
None of the non-Aggregate ones currently defined are, but if in the future one was added, we would want to remove it in this function.